tools/PI/DevHome.PI/Helpers/ExternalTool.cs

```diff
@@ -126,28 +132,51 @@ internal async override void InvokeTool(Window? parentWindow, int? targetProcess
             PInvoke.MessageBox(HWND.Null, errorMessage, _errorTitleText, MESSAGEBOX_STYLE.MB_ICONERROR);
         }
     }
-
-    internal async Task<Process?> InvokeToolInternal(int? pid, HWND? hwnd)
+
+    internal void ValidateOptions(ToolLaunchOptions options)
     {
-        var process = default(Process);
-
-        var parsedArguments = string.Empty;
+        // We can only get stdout if we're launching the process via CreateProcess.
+        if (options.RedirectStandardOut && ActivationType != ToolActivationType.Launch)

```

Comment:
NIT: Change to ActivationType == ToolActivationType.MSIX || ActivationType == ToolActivationType.Protocol.

If another activation type is added the developer needs to know to modify this if statement with the new activation type.

Reply:
It goes either way, right? With the code as-is, if they add a 4th item that could be redirected, then they'd have to update this code. If it couldn't, then they wouldn't need to update the code.

With your suggestion, it would be the opposite. If it couldn't be redirected, they'd need to update the code... if it could, then they could leave it alone.

From what I've seen, it looks like redirection can only happen with a CreateProcess call, so chances are a 4th activation type won't be able to have stdout redirected.... so better to leave the code as-is?